lib/primer/view_components/linters/disallow_component_css_counter.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require_relative "base_linter"
+
+# Load all the other linters so we can filter out their restricted
+# CLASSES—they will be responsible for complaining about the use of
+# those HTML classes.
+Dir[File.join(__dir__, "*.rb")].sort.each do |file|
+  require file unless file == __FILE__
+end
+
+module ERBLint
+  module Linters
+    # Counts the number of times a class reserved for ViewComponents is used
+    class DisallowComponentCssCounter < BaseLinter
+      CLASSES = (
+        JSON.parse(
+          File.read(
+            File.join(__dir__, "..", "..", "..", "..", "static", "classes.json")
+          )
+        ) - BaseLinter.subclasses.reduce([]) do |html_classes, klass|
+          html_classes.concat(klass.const_get(:CLASSES))
+        end
+      ).freeze
+
+      TAGS = nil
+      MESSAGE = "Primer ViewComponents defines some HTML classes with associated styles that should not be used outside those components. (These classes might have their styles changed or even disappear in the future.) Instead of using this class directly, please use its component if appropriate or define the styles you need some other way."
```

Comment:
Should we include what component we think they should be using?

Reply:
Maybe.

We can definitely split up classes based on which components' stylesheets they come from. However, some of our classes are referenced by multiple components. For instance, the `Button--Small` class is referenced by `Primer::Beta::Button`, but it's also used in a selector for `Primer::Alpha::SegmentedControl`. I don't know a good way to determine which component we should suggest. We could suggest all the components that reference a class if you think that would be helpful.

Reply:
hmm, yeah I see the problem. I think this is good enough for now then. 👍🏻